Source/SourceKittenFramework/DyldHelper.swift

```diff
@@ -0,0 +1,192 @@
+//
+//  DyldHelper.swift
+//  sourcekitten
+//
+//  Created by Norio Nomura on 2/11/16.
+//  Copyright © 2016 SourceKitten. All rights reserved.
+//
+
+import Foundation
+
+/// Environment Variable that indicates current process is launched through `execv()`
+private let sourceKittenFrameworkLoadingkey = "SOURCEKITTEN_FRAMEWORK_LOADING"
+
+private let libclangPath = "libclang.dylib"
+private let sourcekitdPath = "sourcekitd.framework/Versions/A/sourcekitd"
+
+/// Checks whether depending dynamic link libraries are loaded or not.
+/// If not, coordinates `DYLD_*` environment variables, calls `execv()` and never returns.
+public func checkDepndenciesOfSourceKittenFramework() {
+    if libraryIsLoaded(libclangPath) && libraryIsLoaded(sourcekitdPath) {
+        return
+    }
+
+    // following codes will be executed when dependencies will not be loaded.
+    // On sourcekitten installation, LC_RPATH of SourceKittenFramework does not contain pathes
+    // which can find libclang.dylib and sourcekitd.framework.
+
+    coordinateEnvironmentVariablesForDYLD()
+    setenv(sourceKittenFrameworkLoadingkey, "1", 1)
+
+    let arguments = NSProcessInfo.processInfo().arguments
+    let path = arguments[0]
+    let argv = CStringArray(arguments)
+    execv(path, argv.pointers) // never returns on normal flow
+    let error = String(UTF8String: strerror(errno))! // swiftlint:disable:this force_unwrapping
+    fatalError("execv failed: \(error)")
```

Comment:
If not, it might be better to follow the way the [stdlib does](https://github.com/apple/swift/blob/master/stdlib/private/SwiftPrivate/SwiftPrivate.swift#L75-L97) instead of CStringArray.